### DISCUSSION TITLE
Reason for changing timestamp format in pretty_ts() function

### DISCUSSION THREAD

**Discussion by perceptron01** [jertel/elastalert2#372] [Q&A] [answered]
In elastalert, `pretty_ts()` function returns timestamp formatted like `'2021-07-27 23:29 JST'`
In elastalert2, however, `pretty_ts()` function returns timestamp formatted like `'2021-07-27 23:29 +0900'`

This is due to changing format string from `'%Z'` to `'%z'`, which is executed at PR #51.
https://github.com/Yelp/elastalert/blob/1dc4f30f30d39a689f419ce19c7e2e4d67a50be3/elastalert/util.py#L205
https://github.com/jertel/elastalert2/blob/722b24fce0fa3272c595e7ea495acdf59b503ee0/elastalert/util.py#L211

I want to know why this change is required, and how I can print timestamp in alert text with the format like `'2021-07-27 23:29 JST'` .

In my use case, I prefer to the former format style.  
But the change described above seems to force us to print timestamp in alert only with the latter one.

**Comment by strikesmash** [comment]
I also traced the conversations and code changes in PR #51.

1. In my understanding of PR #51, changing format string from '%Z' to '%z' does not seem like a necessary change for adding `query_timezone`feature.
So, there should be no reason why the format has to be  '%z'.

2. On the other hand, in the current implementation, the display of the time is fixed to format '%z, and there seems to be no way to display it in format '%Z'.

Since both display methods have their advantages and disadvantages, and each user has different requirements, it would be very useful to be able to select either of them.

I guess whether the format is '%Z' or '%z' has no effect on the behavior of the ElastAlert itself.
Even if there are some oversights or omissions in changing the time display format, we can discuss them in the PR.

**Comment by jertel** [comment, marked as answer]
@dilaverdemirel Could you please comment on why you changed the pretty_ts timezone format in PR #51?

**Reply by dilaverdemirel** [reply]
I suppose, i had have a problem related to correct timezone indicator(for example +03:00). Because Europe/Istanbul zone's indicator changed permanently as +03:00. But some applications didn't work correctly. This change provides to me saw correct indicator. 

But, i think, if you want, it can be like previous style.

**Reply by nsano-rururu** [reply]
Fix util.py.
One of the plans.
The default works with the previous settings. If you have set the environment variables of the OS, use the environment variables. How about solving both problems with this?

before

```
return dt.strftime('%Y-%m-%d %H:%M %z')
```

after

add `import os`

```
return dt.strftime(os.environ.get('PRETTY_TS_FORMAT', '%Y-%m-%d %H:%M %Z'))
```

**Reply by jertel** [reply]
That would solve it! Could also make it a config option, but since the config data isn't available to util.py it would be a bigger change compared to your ENV suggestion.

**Comment by perceptron01** [comment]
@strikesmash @dilaverdemirel @jertel Thank you for your comments! They help me a lot!

Considering @strikesmash's suggestion and also situation like the one @dilaverdemirel has commented above, it would be very useful for pretty_ts() function to allow users to customize timezone format.

I will make a PR in which pretty_ts() and its related components are modified so that users can customize timezone format.

Please give me some time to develop.
